plugins/scenario/World.ts

```diff
@@ -22,13 +22,14 @@ export class World {
     // Q: should we really need to fork/snapshot the deployment manager?
     this.hre = hre;
     this.base = base;
-    this.deploymentManager = new DeploymentManager(base.name, hre, { debug: true });
+    const networkName = base.name === 'development' ? 'development' : base.network;
+    this.deploymentManager = new DeploymentManager(networkName, base.deployment, hre, { debug: true });
     this.snapshotDeploymentManager = this.deploymentManager;
   }
 
   // TODO: Can we do this better?
   isRemoteFork(): boolean {
-    return !!this.base.url;
+    return this.base.network !== 'hardhat';
```

Comment:
If this is better (why) should we get rid of the TODO?

Reply:
deleted